java-checks/src/main/java/org/sonar/java/checks/security/EncryptionAlgorithmCheck.java

```diff
@@ -88,15 +89,20 @@ private static boolean isInsecureAlgorithm(String algorithmName) {
       boolean isRSA = "RSA".equalsIgnoreCase(algorithm);
 
       if ("ECB".equalsIgnoreCase(mode) && !isRSA) {
-        return true;
+        return Optional.of("Use a secure cipher mode.");
       }
-      if ("CBC".equalsIgnoreCase(mode)) {
-        return false;
+
+      if ("CBC".equalsIgnoreCase(mode) && !padding.equalsIgnoreCase("NoPadding")) {
```

Comment:
It's better to be consistent. A string literal should always be on the left or on the right side. Actually, our activated rule insists on the left, so let's keep it that way.